deeprank/generate/DataGenerator.py

```diff
@@ -722,6 +728,89 @@ def add_target(self, prog_bar=False):
         # close the file
         f5.close()
 
+    def realign_complexes(self, align, compute_features=None, pssm_source=None):
+        """Align all the complexes already present in the HDF5.
+        
+        Arguments:
+            align {dict} -- alignement dictionary (see __init__)
+        
+        Keyword Arguments:
+            compute_features {list} -- list of features to be computed
+                                       if None computes the features specified in
+                                       the attrs['features'] of the file (if present)
+             pssm_source {str} -- path of the pssm files. If None the source specfied in
+                                  the attrs['pssm_source'] will be used (if present) (default: {None})
+        
+        Raises:
+            ValueError: If no PSSM detected
+
+        Example :
+
+        >>> database = DataGenerator(hdf5='1ak4.hdf5')
+        >>> # if comute_features and pssm_source are not specified
+        >>> # the values in hdf5.attrs['features'] and hdf5.attrs['pssm_source'] will be used
+        >>> database.realign_complex(align={'axis':'x'}, 
+        >>>                          compute_features['deeprank.features.X'], 
+        >>>                           pssm_source='./1ak4_pssm/')
+        """
+
+        f5 = h5py.File(self.hdf5,'a')
+        
+        mol_names = f5.keys()
+        self.logger.info(f'\n# Start aligning the HDF5 database: {self.hdf5}')
+
+        # deal with the features
+        if self.compute_features is None:
+            if compute_features is None:
+                if 'features' in f5.attrs:
+                    self.compute_features = list(f5.attrs['features'])
+            else:
+                self.compute_features = compute_features
+
+        # deal with the pssm source
+        if self.pssm_source is not None:
+            config.PATH_PSSM_SOURCE = self.pssm_source
+
+        elif pssm_source is not None:
+            config.PATH_PSSM_SOURCE = pssm_source
+        
+        elif 'pssm_source' in f5.attrs:
+            config.PATH_PSSM_SOURCE = f5.attrs['pssm_source']
+        else :
+            raise ValueError('No pssm source detected')
+
+        # loop over the complexes
+        desc = '{:25s}'.format('Add features')
+        for mol in tqdm(mol_names, desc=desc, ncols=100):
+
+            # align the pdb
+            molgrp = f5[mol]
+            pdb = molgrp['complex'][()]
+
+            sqldb = self._get_aligned_sqldb(pdb, align)
+            data = sqldb.sql2pdb()
+
+            data = np.array(data).astype('|S78')
+            molgrp['complex'][...] = data
+
+            # remove prexisting features
```

Comment:
`mapped_features` should also be removed here to avoid that users forget to do re-mapping.

Reply:
true

Reply:
All the other parts of this PR looks good, you can merge it then. Thanks!